### PR TITLE
Add missing rgb matrix default parameters

### PR DESCRIPTION
**Pull request by FabienFellay**
This PR proposes to add the missing configurable rgb matrix default parameters in `quantum/rgb_matrix/rgb_matrix.c`.

## Description

`quantum/rgb_matrix/rgb_matrix.c` offers the possibility of configuring the following rgb matrix parameters in a `config.h` file:
* `RGB_MATRIX_DEFAULT_MODE`
* `RGB_MATRIX_DEFAULT_HUE`
* `RGB_MATRIX_DEFAULT_SAT`
* `RGB_MATRIX_DEFAULT_VAL`
* `RGB_MATRIX_DEFAULT_SPD`

However, there is no way of configuring the default `rgb_matrix_config` flags nor the default `rgb_matrix_config` enable-disable status. Therefore, the following `config.h` options have been added:
* `RGB_MATRIX_DEFAULT_STATUS`
* `RGB_MATRIX_DEFAULT_FLAG`

For example, this will allow to directly configure the following default rgb matrix option (after an EEPROM reset) in a `config.h` file:

```c
/* Select the default RGB matrix flag (disable RGB effect by default) */
#define RGB_MATRIX_DEFAULT_FLAG LED_FLAG_NONE
```

Currently, I have to rely on the use of `keyboard_post_init_user()` in a `keymap.c` file:

```c
void keyboard_post_init_user(void) {
    // Use non noeeprom versions to write to EEPROM as well
    rgb_matrix_set_flags(LED_FLAG_NONE);  // Select the default RGB matrix flag (disable RGB effect by default)
}
```

However, this applies at each boot (and not only at each EEPROM reset, as wished). Additionally, it unnecessarily causes multiple write to the EEPROM. This could be avoided by directly specifying the desired (not supported yet) default options in a `config.h`. Then, `eeconfig_update_rgb_matrix_default()` will directly take care of them.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

I am not aware of an issue opened for that.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage, but not sure of all the consequences of the changes, though).

### CODE REVIEW

**Comment by FabienFellay** [comment]
@lesshonor Thank you for your suggestion. Using `void eeconfig_init_user(void)` was actually my first attempt, but when I do the following in my `keymap.c`:

```c
void eeconfig_init_user(void) {
    // Use non noeeprom versions to write to EEPROM as well
    rgb_matrix_set_flags(LED_FLAG_NONE);  // Select the default RGB matrix flag (disable RGB effect by default)
}
```

It just does not work: all the default parameters set by `quantum/rgb_matrix/rgb_matrix.c`:

```c
void eeconfig_update_rgb_matrix_default(void) {
    dprintf("eeconfig_update_rgb_matrix_default\n");
    rgb_matrix_config.enable = 1;
    rgb_matrix_config.mode   = RGB_MATRIX_DEFAULT_MODE;
    rgb_matrix_config.hsv    = (HSV){RGB_MATRIX_DEFAULT_HUE, RGB_MATRIX_DEFAULT_SAT, RGB_MATRIX_DEFAULT_VAL};
    rgb_matrix_config.speed  = RGB_MATRIX_DEFAULT_SPD;
    rgb_matrix_config.flags  = LED_FLAG_ALL;
    eeconfig_flush_rgb_matrix(true);
}
```

seems to override what I do in `void eeconfig_init_user(void)` (and thus the rgb_matrix part of the example of https://docs.qmk.fm/#/feature_eeprom?id=eeconfig-function-documentation is not working for me). This is the only thing that motivated this PR, actually.

However, if you have ideas of why `void eeconfig_init_user(void)` is not working when setting rgb_matrix default parameters, I will agree this PR is not needed.

Finally, what I meant by *avoiding multiple write to the EEPROM* is that, by using the proposed new preprocessor #define directive, the rgb matrix defaults are only written once to the EEPROM by `quantum/rgb_matrix/rgb_matrix.c`. The example of https://docs.qmk.fm/#/feature_eeprom?id=eeconfig-function-documentation, if it was working, seems to cause multiple write. The goal was just to minimize the number of write to the EEPROM (but honestly, I admit that is too much care and not necessary).

Can you confirm that

```c
void eeconfig_init_user(void) {
    // Use non noeeprom versions to write to EEPROM as well
    rgb_matrix_set_flags(LED_FLAG_NONE);  // Select the default RGB matrix flag (disable RGB effect by default)
}
```

has no effect? Or am I missing something? It does have an effect in `void keyboard_post_init_user(void)` but I want to retain my setting on reboot. I only want to set custom defaults on EEPROM reset.

(*sorry for the `master` targeting instead of `develop`*)

**Comment by FabienFellay** [comment]
OK: on the `develop` branch, it seems they somehow introduced something similar to the missing default parameters I proposed in this PR. So the latter definitely may not be needed anymore. However, you could maybe still help me to figure out what is wrong with `void eeconfig_init_user(void)`?

**Comment by FabienFellay** [comment]
I am customizing a keychron q6 keyboard based on what Keychron released here: `qmk_firmware/keyboards/keychron/q6/iso_encoder/keymaps/keychron/`. When host indicator #define are present (numlock or caps locks), which is my case, they somehow use the flag integer instead of `rgb_matrix_enable()`, `rgb_matrix_disable()` and `rgb_matrix_toggle()` in order to turn RGB effect on and off while preserving the indicator state.

You can look at what they did in `keyboards/keychron/q6/q6.c`. I want to stick with this logic (which works well) and the ability to set the flag state after an EEPROM reset would do exactly what I expect (want). This is a typical use case where modifying the default flags would be useful.

**Comment by FabienFellay** [comment]
I resolved the conflicts that occurred by changing the target branch. In short:

* Now, the only *new* proposed default parameter that can be specified in a `config.h` is `RGB_MATRIX_DEFAULT_FLAG` because the `develop` branch already implemented `RGB_MATRIX_DEFAULT_ON` (that I initially proposed as well under the less clear name `RGB_MATRIX_DEFAULT_STATUS`).
* The documentation `docs/feature_rgb_matrix.md` has been updated accordingly (and the order of options has been fixed to be consistent with the order in the code in `quantum/rgb_matrix/rgb_matrix.c` and `quantum/rgb_matrix/rgb_matrix.h`).

**Comment by FabienFellay** [comment]
I just tested this on a quite recent state of the `develop` branch and I can confirm that it is working as expected and that it does not break anything.

It would allow one to now put the following in a `config.h` file:

```c
#pragma once

#define RGB_MATRIX_DEFAULT_ON true // Sets the default enabled state, if none has been set
#define RGB_MATRIX_DEFAULT_FLAG LED_FLAG_NONE // Sets the default LED flags, if none has been set
```

which is useful to initially (after a EEPROM reset only) power off the rgb effect while preserving the host indicators LEDs (caps lock, num lock) when using a Keychron q6 keyboard (see `keyboards/keychron/q6/q6.c`).

**Comment by FabienFellay** [comment]
Hello guys, the github-actions bot wants to close this PR as there has been no feedback for 45 days, apparently. As a reminder, this PR is ready, and its status should be `awaiting review`.

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -873,12 +873,13 @@ These are defined in [`color.h`](https://github.com/qmk/qmk_firmware/blob/master
...
 #define RGB_MATRIX_LED_FLUSH_LIMIT 16 // limits in milliseconds how frequently an animation will update the LEDs. 16 (16ms) is equivalent to limiting to 60fps (increases keyboard responsiveness)
 #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 200 // limits maximum brightness of LEDs to 200 out of 255. If not defined maximum brightness is set to 255
+#define RGB_MATRIX_DEFAULT_ON true // Sets the default enabled state, if none has been set
 #define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT // Sets the default mode, if none has been set
 #define RGB_MATRIX_DEFAULT_HUE 0 // Sets the default hue value, if none has been set
 #define RGB_MATRIX_DEFAULT_SAT 255 // Sets the default saturation value, if none has been set
-#define RGB_MATRIX_DEFAULT_ON true // Sets the default enabled state, if none has been set
 #define RGB_MATRIX_DEFAULT_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS // Sets the default brightness value, if none has been set
 #define RGB_MATRIX_DEFAULT_SPD 127 // Sets the default animation speed, if none has been set
+#define RGB_MATRIX_DEFAULT_FLAG LED_FLAG_ALL // Sets the default LED flags, if none has been set
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md:882]:
```suggestion
#define RGB_MATRIX_DEFAULT_FLAGS LED_FLAG_ALL // Sets the default LED flags, if none has been set
```

File: quantum/rgb_matrix/rgb_matrix.c [hunk 2]
```diff
@@ -100,7 +100,7 @@ void eeconfig_update_rgb_matrix_default(void) {
     rgb_matrix_config.mode   = RGB_MATRIX_DEFAULT_MODE;
     rgb_matrix_config.hsv    = (HSV){RGB_MATRIX_DEFAULT_HUE, RGB_MATRIX_DEFAULT_SAT, RGB_MATRIX_DEFAULT_VAL};
     rgb_matrix_config.speed  = RGB_MATRIX_DEFAULT_SPD;
-    rgb_matrix_config.flags  = LED_FLAG_ALL;
+    rgb_matrix_config.flags  = RGB_MATRIX_DEFAULT_FLAG;
```
Comment (fauxpark) [inline quantum/rgb_matrix/rgb_matrix.c:103]:
```suggestion
    rgb_matrix_config.flags  = RGB_MATRIX_DEFAULT_FLAGS;
```

File: quantum/rgb_matrix/rgb_matrix.h [hunk 3]
```diff
@@ -78,6 +78,10 @@
 #    define RGB_MATRIX_DEFAULT_SPD UINT8_MAX / 2
 #endif
 
+#ifndef RGB_MATRIX_DEFAULT_FLAG
+#    define RGB_MATRIX_DEFAULT_FLAG LED_FLAG_ALL
```
Comment (fauxpark) [inline quantum/rgb_matrix/rgb_matrix.h:82]:
```suggestion
#ifndef RGB_MATRIX_DEFAULT_FLAGS
#    define RGB_MATRIX_DEFAULT_FLAGS LED_FLAG_ALL
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by FabienFellay** [comment]
Thank you for your review, the suggestions have been applied.

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**